ApplensBackend/Helpers/DateTimeHelper.cs

```diff
@@ -0,0 +1,149 @@
+        {
+            timeGrainTimeSpan = KustoConstants.DefaultTimeGrain;
+            bool result = true;
+            errorMessage = string.Empty;
+
+            result = PrepareStartEndTimeUtc(startTime, endTime, timeWindowInDays, out startTimeUtc, out endTimeUtc, out errorMessage);
+
+            startTimeUtc = GetDateTimeInUtcFormat(RoundDownTime(startTimeUtc, timeGrainTimeSpan));
+            endTimeUtc = GetDateTimeInUtcFormat(RoundDownTime(endTimeUtc, timeGrainTimeSpan));
+
+            if (startTimeUtc == endTimeUtc)
+            {
+                endTimeUtc = startTimeUtc.Add(TimeSpan.FromDays(timeWindowInDays));
+            }
+
+            return result;
+        }
+
+        internal static bool PrepareStartEndTimeUtc(string startTime, string endTime, int timeWindowInDays, out DateTime startTimeUtc, out DateTime endTimeUtc, out string errorMessage)
+        {
+            //1. no startTime, no endTime => return current time - 24 hours, current time
+            //2. startTime, no endTime => return start time, end time = start time + 24 hours
+            //3. no startTime, endTime => return start time = end time - 24 hours, end time
+            //4. startTime, endTime => return start time, end time
+
+            DateTime currentUtcTime = GetDateTimeInUtcFormat(DateTime.UtcNow);
+            bool result = true;
+            errorMessage = string.Empty;
+
+            if (string.IsNullOrWhiteSpace(startTime) && string.IsNullOrWhiteSpace(endTime))
+            {
+                endTimeUtc = currentUtcTime;
+                startTimeUtc = endTimeUtc.AddDays(-(timeWindowInDays));
+            }
+            else if (string.IsNullOrWhiteSpace(startTime))
+            {
+                result = ParseDateTimeParameter("endTime", endTime, currentUtcTime, out endTimeUtc);
+                startTimeUtc = endTimeUtc.AddDays(-(timeWindowInDays));
+            }
+            else if (string.IsNullOrWhiteSpace(endTime))
+            {
+                result = ParseDateTimeParameter("startTime", startTime, currentUtcTime.AddDays(-(timeWindowInDays)), out startTimeUtc);
+                endTimeUtc = startTimeUtc.AddDays(timeWindowInDays);
+                if (endTimeUtc > currentUtcTime)
+                {
+                    endTimeUtc = currentUtcTime;
+                }
+            }
+            else
+            {
+                result = ParseDateTimeParameter("endTime", endTime, currentUtcTime, out endTimeUtc);
+                result &= ParseDateTimeParameter("startTime", startTime, currentUtcTime.AddDays(-(timeWindowInDays)), out startTimeUtc);
+            }
+
+            if (result == false)
+            {
+                errorMessage = "Cannot parse invalid date time. Valid Time format is yyyy-mm-ddThh:mm";
+                return false;
+            }
+
+            if (startTimeUtc > endTimeUtc)
```

Comment:
If the endTimeUtc is later than current UTC, will it be set to be current UTC?

Reply:
yes, exactly. if the Endtime is some future date time, it will set to current time